src/CachePeer.cc

```diff
@@ -109,6 +112,25 @@ CachePeer::connectTimeout() const
     return Config.Timeout.peer_connect;
 }
 
+CachePeerProbeCodeContext::CachePeerProbeCodeContext(CachePeer *p) : peer(p) {}
+
+ScopedId
+CachePeerProbeCodeContext::codeContextGist() const
+{
+    if (peer.valid())
+        return peer->id.detach();
+
+    return ScopedId("CachePeerProbeCodeContext w/o peer");
```

Comment:
Let's simplify this code and not add CachePeer::id in _this_ PR. We already have a TODO about adding InstanceId to such classes. Also, the whole idea of having codeContextGist() returning an identifier may be an API bug: We should rethink how FadingCodeContext works, so that we can avoid identifier copying and convert this method into an std::ostream-printing method.

```suggestion
    // Unfortunately, peer->name lifetime is too short. See also: AnyP::PortCfg::codeContextGist().
    return ScopedId("cache_peer probe");
```